File: annotationprocessortesting/src/main/java/com/tickaroo/tikxml/regressiontests/paths/Boss.java
package com.tickaroo.tikxml.regressiontests.paths;

import com.tickaroo.tikxml.annotation.PropertyElement;
import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class Boss extends Person {
  @PropertyElement String name;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Boss)) return false;
    if (!super.equals(o)) return false;

    Boss boss = (Boss) o;

    return name != null ? name.equals(boss.name) : boss.name == null;
  }

  @Override public int hashCode() {
    int result = super.hashCode();
    result = 31 * result + (name != null ? name.hashCode() : 0);
    return result;
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/regressiontests/paths/Boss.kt
package com.tickaroo.tikxml.regressiontests.paths

import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
class Boss : Person() {
    @PropertyElement
    var name: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Boss) return false
        if (!super.equals(other)) return false

        val boss = other as Boss?

        return if (name != null) name == boss!!.name else boss!!.name == null
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + if (name != null) name!!.hashCode() else 0
        return result
    }
}
